lib/lrama/grammar/parameterizing_rule_resolver.rb

```diff
@@ -0,0 +1,31 @@
+module Lrama
+  class Grammar
+    class ParameterizingRuleResolver
+      attr_reader :rules, :tokens, :term
+
+      def initialize
+        @rules = []
+        @tokens = []
+        @term = nil
+        @parameterizing_rule_builders = []
+      end
+
+      def add_parameterizing_rule_builder(builder)
+        @parameterizing_rule_builders << builder
+      end
+
+      def defined?(name)
+        @parameterizing_rule_builders.any? { |builder| builder.name == name }
+      end
+
+      def build_rules(token, rule_counter, lhs_tag, user_code, precedence_sym, line)
+        @parameterizing_rule_builders.each do |builder|
```

Comment:
It may be needed to filter `@parameterizing_rule_builders` to only single builder whose name matches with `token.s_value` otherwise rules are generated for all `@parameterizing_rule_builders`.
Considering about the case below:

```
%rule defined_option(X): /* empty */
                       | X
                       ;

%rule defined_option2(X): /* empty */
                        | X
                        ;

%%

program         : defined_option(number) <i>
                ;
```

Reply:
Thank you very much! The following commit fixes the problem:
https://github.com/ruby/lrama/pull/285/commits/b07ce82bef231fe02290ea480ee1371411c2c8d8